### PR TITLE
secure ipfs gateway

### PR DESCRIPTION
**Pull request by pldespaigne**
Hello guys 👋 @makoto @Arachnid @jefflau 

I have updated the code to resolve ipfs hash trough a secure gateway.
This is a huge security improvement.

This PR has a dependency on this other PR 👇 

Everything is detailed on the other [PR](https://github.com/ensdomains/ui/pull/46)

As always thanks a lot for your time and your work!

PS : you can read more about secure ipfs in this [MetaMask PR](https://github.com/MetaMask/metamask-extension/pull/7362)

### CODE REVIEW

File: src/components/Links/ContentHashLink.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ const ContentHashLink = ({ value, contentType }) => {
     return <DecodedError>{error}</DecodedError>
   }
   if (protocolType === 'ipfs') {
```
Comment (lidel) [inline src/components/Links/ContentHashLink.js:41]: `https://dweb.link/ipfs/${decoded}` will also work (will redirect to correct subdomain + convert CID to DNS-compatible encoding, if needed)

File: src/components/Links/ContentHashLink.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ const ContentHashLink = ({ value, contentType }) => {
...
-    externalLink = `https://gateway.ipfs.io/ipfs/${decoded}`
+    externalLink = `https://${decoded}.ipfs.dweb.link` // using ipfs's secured origin gateway
```
Comment (lidel) [inline src/components/Links/ContentHashLink.js:42]: `https://dweb.link/ipfs/${decoded}` will also work (will redirect to correct subdomain + convert CID to DNS-compatible encoding, if needed)

Comment (lidel) [inline src/components/Links/ContentHashLink.js:42]:
```suggestion
    externalLink = `https://dweb.link/ipfs/${decoded}` // using ipfs's secured origin gateway
```

We recommend keeping the path-based router for improved compatibility.
go-ipfs 0.5+ will redirect to correct subdomain, and go-ipfs 0.7+ will convert CID to DNS-compatible encoding, if needed

**Review by lidel — CHANGES_REQUESTED** [review]
@pldespaigne see comment inline

**Comment by makoto** [comment]
Duplicate to https://github.com/ensdomains/ens-app/pull/936

**Pull request closed without merging**